### ISSUE TITLE
Missing targets when phonyOnlyTargets is true

### ISSUE THREAD

**Issue by codefrogs** [microsoft/vscode-makefile-tools#574]
Let's say I have in settings.json:
`"makefile.phonyOnlyTargets": "true"`

And I have the following makefile:
```
all: hello list fake
list:
	@echo "List"
hello:
	@echo "Hello"
fake: .
	@echo "Fake target"
.PHONY: all list hello fake
```
I then click on: 'Build target' pencil icon.
The 'quick pick' then only lists: 'all', and 'fake'.

Should the targets: 'hello', and 'list' not also be listed?
Are only phony targets with a dependency listed?

**Comment by Yingzi1234** [comment]
@gcampbell-msft  
We've triaged this issue as "enhancement", please feel free if we triaged incorrectly. Thanks!
The user's issue is, after explicitly declaring the. PHONY target and adding pseudo-targets to the "Makefile" file that represent running specific commands, is it possible to list these pseudo-targets in the "Makefile: Setting up targets built by make" list?
![image](https://github.com/microsoft/vscode-makefile-tools/assets/101234795/fa60c61d-4e73-4708-a95d-1d33648aedb9)

**Comment by gcampbell-msft** [comment]
@codefrogs Thanks for the issue! @Yingzi1234 I actually think that this is a bug, so I'll change the label, but thank you for addressing this and giving context.

**Comment by codefrogs** [comment]
![makefile-tools-2024-05-28](https://github.com/microsoft/vscode-makefile-tools/assets/79409528/a35a51da-353a-4279-bf21-8d1eb8fa9a59)
In settings.json:  "makefile.phonyOnlyTargets": true
Phony targets without a pre-requisite are not listed in the build targets.

@gcampbell-msft Thanks for getting back to me. Perhaps I'm doing something wrong? But I've tried pre-release 10.8 on Ubuntu, and still have the same issue.
@Yingzi1234 Yes, I can. Perhaps I haven't updated something else?

**Comment by gcampbell-msft** [comment]
@codefrogs Hm, maybe I need to attempt to reproduce on Linux, on Windows, it seems to behave as expected. I'll do that as soon as I can. 

@Yingzi1234 If you have time, please attempt to reproduce in Linux. Thanks!

**Comment by codefrogs** [comment]
@gcampbell-msft @Yingzi1234 I'm testing within a Ubuntu 22.04.4 LTS VirtualBox instance hosted on Windows. The VM has one processor, and only 2GB memory. Hope that helps.

**Comment by Yingzi1234** [comment]
@gcampbell-msft @codefrogs This issue has been fixed in the latest versions for Windows, Mac and Linux platforms, as detailed below.

**ENV:**
VScode: 1.89.1
Makefile tools: v0.10.9(pre-release)
C/C++: v1.20.5

**Repro steps:**
1. Clone a repo https://github.com/remonbonbon/makefile-example.git with VScode
2. Replace the contents of file "Makefile" file with the following code and save it
```
all: hello list fake
list:
	@echo "List"
hello:
	@echo "Hello"
fake: .
	@echo "Fake target"
.PHONY: all list hello fake
```
3. Go to 'Makefile' window to set the 'Build target'

**Actual result:**
The list of build targets will appear with the content you just added.

Windows:
![574](https://github.com/microsoft/vscode-makefile-tools/assets/101234795/545d9e0e-2150-47e5-ba23-04cee5674d0f)

Mac；
![image](https://github.com/microsoft/vscode-makefile-tools/assets/101234795/a423b8b5-49c0-4fa5-9a02-a8c3b9ea9142)

Linux：
![image](https://github.com/microsoft/vscode-makefile-tools/assets/101234795/2603edf3-4246-45ee-9122-16fcefbb4f7d)

**Comment by codefrogs** [comment]
@Yingzi1234 Good job! I've repeated your 'Repro steps' in a new VM, and can confirm this issue has been resolved with the pre-release 0.10.9.
